webrender/res/ps_text_run.glsl

```diff
@@ -61,64 +61,82 @@ TextRun fetch_text_run(int address) {
     return TextRun(data[0], data[1], data[2].xy);
 }
 
-VertexInfo write_text_vertex(vec2 clamped_local_pos,
-                             RectWithSize local_clip_rect,
+VertexInfo write_text_vertex(RectWithSize local_clip_rect,
                              float z,
                              Transform transform,
                              PictureTask task,
                              vec2 text_offset,
-                             RectWithSize snap_rect,
+                             vec2 glyph_offset,
+                             RectWithSize glyph_rect,
                              vec2 snap_bias) {
-    // Transform the current vertex to world space.
-    vec4 world_pos = transform.m * vec4(clamped_local_pos, 0.0, 1.0);
-
-    // Convert the world positions to device pixel space.
-    float device_scale = uDevicePixelRatio / world_pos.w;
-    vec2 device_pos = world_pos.xy * device_scale;
+    // The offset to snap the glyph rect to a device pixel
     vec2 snap_offset = vec2(0.0);
 
-#if defined(WR_FEATURE_GLYPH_TRANSFORM)
-    bool remove_subpx_offset = true;
-#else
-    bool remove_subpx_offset = transform.is_axis_aligned;
-#endif
+#ifndef WR_FEATURE_GLYPH_TRANSFORM
     // Compute the snapping offset only if the scroll node transform is axis-aligned.
-    if (remove_subpx_offset) {
+    if (transform.is_axis_aligned) {
```

Comment:
can we please not change the block structure/indentation by the preprocessor branches? let's have a boolean to check in `if` uniformly

Reply:
We can always just have stuff declared but not defined *outside* of the block, and only used inside:
```glsl
vec4 some_value;
if (condition) {
   some_value = xxx;
}
if (condition) {
   yyy = some_value;
}
```
The compiler can see that we aren't using uninitialized variables, and we don't have to sacrifice the block structure/control flow for it.